### ISSUE TITLE
[Feature Request] disable services in the configuration

### ISSUE THREAD

**Issue by speednroll** [TeamNewPipe/NewPipe#1116]
I love that Newpipe has the possibility of using multiple services, but what happens if someone prefers to use certain services and not others? It would be nice to add a section in the configuration to disable services

**Comment by justanidea** [comment]
Or to download only the section you want, to make the app as light as possible :)

**Comment by theScrabi** [comment]
Yes when there are more than only two services this is something we can consider.

**Comment by mhmdanas** [comment]
NewPipe currently has five services, so I think this is worth re-visiting.

**Comment by FletcherD** [comment]
I think that this is fairly necessary, because we currently cannot use the Bandcamp cart or buy anything without this.

**Comment by hermitical** [comment]
This should definitely be given attention. I do not wish to use NewPipe for Bandcamp, this leads to NewPipe hijacking links and literally makes it impossible to access Bandcamp properly when going through a browser (as you annoyingly have to for some things on Bandcamp)

**Comment by mhmdanas** [comment]
@FletcherD @hermitical what Android versions do you have? In Android 12, you are able to enable certain links for any given apps, while keeping the others disabled. For instance, you can make NewPipe handle youtube.com links, but not *.bandcamp.com ones. I don't know if you can do something similar in earlier Android versions though, so you'll have to do some research.

**Comment by Wyntilda** [comment]
While this is a setting in newer versions of Android, NewPipe doesn't seem to restrict itself to just the allowed links. I only allow YouTube links for NewPipe and it still opens Bandcamp links. There doesn't seem to be a way to disable this from happening.

**Comment by opusforlife2** [comment]
@Wyntilda In Android 12+ you can select exactly which links you want apps to be able to open from a list.

**Comment by Wyntilda** [comment]
@opusforlife2 I'm aware; what I'm saying is that Bandcamp is not on the allowed links list for NewPipe, only YouTube links are, yet NewPipe still opens Bandcamp links.

**Comment by opusforlife2** [comment]
Oh wow. And clearing defaults for link opening settings has no effect on this? Do you have the Bandcamp app installed? Have you tried Firefox, if you're using Chrome?

**Comment by PigDan20** [comment]
> This should definitely be given attention. I do not wish to use NewPipe for Bandcamp, this leads to NewPipe hijacking links and literally makes it impossible to access Bandcamp properly when going through a browser (as you annoyingly have to for some things on Bandcamp)

This same thing is happening to me. It's very, very annoying and makes navigating certain webpages or apps impossible due to the URL opening feature. The Android app setting adjustments do absolutely nothing to turn this off from a user's perspective, so NewPipe is causing the issues likely. Like what was said earlier, any Bandcamp URL opened on a Bandcamp webpage in Firefox (or I assume anywhere else in Android) automatically opens up the NewPipe overlay, whether that's a link to a musician's albums or elsewhere.
Any suggestions to make this go away, or plans from the devs to add proper customization to optionally turn off URL opening altogether? On Android 13 now.

**Comment by fynngodau** [comment]
@opusforlife2 On my Android 13 device, turning off the "*.bandcamp.com" URL for NewPipe in the system settings has the desired effect, i.e. NewPipe no longer opens when clicking links elsewhere in the system, and after adding the link again, NewPipe is always opened (note that there is no way to have a picker for this natively anymore, unfortunately). It is unclear to me why this does not work for the other commenters.

In any case, I believe that NewPipe can not solve the problem because the links that and app handles are hardcoded in the manifest and can't be changed at runtime. The only option I see would be to open the browser from within NewPipe automatically for some links if the user wishes so – which I consider is a not-so-sensical option, because we really wouldn't want _every_ app to implement this, as figuring out which app should handle an intent / a link is a task that belongs to the Android system.

**Comment by PigDan20** [comment]
> @opusforlife2 On my Android 13 device, turning off the "*.bandcamp.com" URL for NewPipe in the system settings has the desired effect, i.e. NewPipe no longer opens when clicking links elsewhere in the system, and after adding the link again, NewPipe is always opened (note that there is no way to have a picker for this natively anymore, unfortunately). It is unclear to me why this does not work for the other commenters.
> 
> In any case, I believe that NewPipe can not solve the problem because the links that and app handles are hardcoded in the manifest and can't be changed at runtime. The only option I see would be to open the browser from within NewPipe automatically for some links if the user wishes so – which I consider is a not-so-sensical option, because we really wouldn't want _every_ app to implement this, as figuring out which app should handle an intent / a link is a task that belongs to the Android system.

Are you referring to this screen or a different one in the Android 13 setting menu?
![Screenshot_20230204_164757_Settings](https://user-images.githubusercontent.com/77480742/216790917-6b814965-d2c3-4f9e-93f2-763b411733ad.jpg)

**Comment by fynngodau** [comment]
@PigDan20 Indeed, this is the settings page I am referring to.

Your Android flavor seems to not be AOSP-like, and I suspect this to be a bug that your device manufacturer is responsible for. Could you specify your device model, please?

**Comment by PigDan20** [comment]
> @PigDan20 Indeed, this is the settings page I am referring to.
> 
> Your Android flavor seems to not be AOSP-like, and I suspect this to be a bug that your device manufacturer is responsible for. Could you specify your device model, please?

Thanks for the reply, appreciated. Sure (and yeah, it does seem like a bug)!
![Screenshot_20230205_213709_Settings](https://user-images.githubusercontent.com/77480742/216870764-ed7d6a47-6030-4013-80bf-bca3adfa55cf.jpg)

**Comment by fynngodau** [comment]
@PigDan20 I was only able to test on a Samsung Galaxy S22 Ultra with Android 12, not Android 13. On this device, the described behavior was not the case, and both unchecking "*.bandcamp.com" under Supported web addresses, as well as disabling the "Open supported links" toggle entirely, both caused links to no longer be opened in NewPipe.

**Comment by PigDan20** [comment]
> @PigDan20 I was only able to test on a Samsung Galaxy S22 Ultra with Android 12, not Android 13. On this device, the described behavior was not the case, and both unchecking "*.bandcamp.com" under Supported web addresses, as well as disabling the "Open supported links" toggle entirely, both caused links to no longer be opened in NewPipe.

Right. Strange, yeah that's the thing I was testing earlier. But I did this again -- actually both are automatically turned off -- but I enabled it and disabled both and this still happens.
